evalutils/evalutils.py

```diff
@@ -158,10 +180,158 @@ def process_case(self, *, idx: int, case: DataFrame) -> Dict:
         return {}
 
     def save(self):
-        with open(self._output_file, "w") as f:
+        with open(str(self._output_file), "w") as f:
             json.dump(self._case_results, f)
 
 
+class Algorithm(BaseAlgorithm):
```

Comment:
Is `BaseAlgorithm` used anywhere else but here? If not, why not combine the classes?

Reply:
Yes, that would be much better. I'll do that.